src/test/java/io/jenkins/plugins/analysis/warnings/recorder/RecorderITest.java

```diff
@@ -0,0 +1,145 @@
+package io.jenkins.plugins.analysis.warnings.recorder;
+
+import org.junit.Test;
+
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.steps.IssuesRecorder;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerSuite;
+import io.jenkins.plugins.analysis.core.util.QualityGate.QualityGateResult;
+import io.jenkins.plugins.analysis.core.util.QualityGate.QualityGateType;
+import io.jenkins.plugins.analysis.core.util.QualityGateStatus;
+import io.jenkins.plugins.analysis.warnings.Java;
+import io.jenkins.plugins.analysis.warnings.recorder.pageobj.InfoPage;
+
+import static io.jenkins.plugins.analysis.core.assertions.Assertions.*;
+
+public class RecorderITest extends IntegrationTestWithJenkinsPerSuite {
+
+    @Test
+    public void shouldCreateSuccessfulFreestyleJobWithWarnings() {
+        FreeStyleProject project = createFreeStyleProject();
+
+        Java java = new Java();
+        java.setPattern("**/*.txt");
+
+        IssuesRecorder recorder = enableWarnings(project, java);
+        recorder.addQualityGate(5, QualityGateType.TOTAL, QualityGateResult.FAILURE);
+
+        copySingleFileToWorkspace(project, "../javac.txt", "javac.txt");
+
+        AnalysisResult result = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+
+        assertThat(result).hasTotalSize(2);
+        assertThat(result).hasQualityGateStatus(QualityGateStatus.PASSED);
+    }
+
+    @Test
+    public void shouldCreateHealthReportOf100() {
+        FreeStyleProject project = createFreeStyleProject();
+        AnalysisResult result = builtResultWithWarnings(project, "java0Warnings.txt");
+        assertThat(result).hasTotalSize(0);
+        assertThat(project.getBuildHealth().getScore()).isEqualTo(100);
+    }
+
+    @Test
+    public void shouldCreateHealthReportOf90() {
+        FreeStyleProject project = createFreeStyleProject();
+        AnalysisResult result = builtResultWithWarnings(project, "java1Warnings.txt");
+        assertThat(result).hasTotalSize(1);
+        assertThat(project.getBuildHealth().getScore()).isEqualTo(90);
+    }
+
+    @Test
+    public void shouldCreateHealthReportOf10() {
+        FreeStyleProject project = createFreeStyleProject();
+        AnalysisResult result = builtResultWithWarnings(project, "java9Warnings.txt");
+        assertThat(result).hasTotalSize(9);
+        assertThat(project.getBuildHealth().getScore()).isEqualTo(10);
+    }
+
+    @Test
+    public void shouldCreateHealthReportOf0() {
+        FreeStyleProject project = createFreeStyleProject();
+        AnalysisResult result = builtResultWithWarnings(project, "java10Warnings.txt");
+        assertThat(result).hasTotalSize(10);
+        assertThat(project.getBuildHealth().getScore()).isEqualTo(0);
+    }
+
```

Comment:
Die obigen Tests schauen recht gleich aus, das lässt sich noch mehr extrahieren in Methoden